data-service/src/test/java/com/amazonaws/blox/dataservice/repository/ListClustersTest.java

```diff
@@ -0,0 +1,104 @@
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.groups.Tuple.tuple;
+
+import com.amazonaws.blox.dataservice.model.EnvironmentId;
+import com.amazonaws.blox.dataservice.test.data.ModelBuilders;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+public class ListClustersTest extends EnvironmentRepositoryTestCase {
+  private static final String CLUSTER = "ClusterOne";
+  private static final String OTHER_CLUSTER = "ClusterTwo";
+  private static final String ENVIRONMENT_NAME = "EnvironmentOne";
+  private static final String OTHER_ENVIRONMENT_NAME = "EnvironmentTwo";
+  ModelBuilders models = ModelBuilders.builder().build();
+
+  List<EnvironmentId> ids =
+      Arrays.asList(
+          models.environmentId("1", "ClusterOne", "EnvironmentOne"),
+          models.environmentId("2", "ClusterTwo", "EnvironmentTwo"),
+          models.environmentId("2", "ClusterThree", "EnvironmentThree"));
+
+  public void givenEnvironments(List<EnvironmentId> ids) throws Exception {
+    for (EnvironmentId id : ids) {
+      repo.createEnvironmentAndEnvironmentRevision(
+          models.environment().environmentId(id).build(),
+          models.environmentRevision().environmentId(id).build());
+    }
+  }
+
+  public void givenEnvironments(EnvironmentId... ids) throws Exception {
+    givenEnvironments(Arrays.asList(ids));
+  }
+
+  @Test
+  public void itDoesntIncludeDuplicateClusters() throws Exception {
+    givenEnvironments(ids);
+    givenEnvironments(
+        models.environmentId(
+            ids.get(0).getAccountId(), ids.get(0).getCluster(), "OtherEnvironment"));
+
+    assertThat(repo.listClusters(ids.get(0).getAccountId(), ids.get(0).getCluster()))
+        .extracting("accountId", "clusterName")
+        .containsExactlyInAnyOrder(tuple(ids.get(0).getAccountId(), ids.get(0).getCluster()));
+  }
+
+  @Test
+  public void itFiltersClustersByAccountIdIfGiven() throws Exception {
+    givenEnvironments(ids);
+
+    assertThat(repo.listClusters(ids.get(0).getAccountId(), null))
+        .extracting("accountId", "clusterName")
+        .containsExactlyInAnyOrder(tuple(ids.get(0).getAccountId(), ids.get(0).getCluster()));
+  }
+
+  @Test
+  public void itFiltersClustersByAccountIdAndClusterPrefixIfGiven() throws Exception {
+    givenEnvironments(ids);
+
+    assertThat(repo.listClusters(ids.get(1).getAccountId(), "ClusterT"))
```

Comment:
How about directly use the value instead of `ids.get(1).getAccountId()`?    like`repo.listClusters("1", "Cluster")`.  Otherwise, we have to go back and read the test data to know what's the value it use. 

Or even we could name the account name as "account_id_with_two_clusters" to have more readable code.

Reply:
I like it! Makes sense.

Reply:
I'm thinking something like this for the test cases:

```java
  @Test
  public void itDoesntIncludeDuplicateClusters() throws Exception {
    EnvironmentIdBuilder sameCluster = models.environmentId().accountId("1").cluster("TestCluster");
    givenEnvironments(
        sameCluster.environmentName("EnvironmentOne").build(),
        sameCluster.environmentName("EnvironmentTwo").build());

    assertThat(repo.listClusters("1", "TestCluster"))
        .extracting("accountId", "clusterName")
        .containsExactlyInAnyOrder(tuple("1", "TestCluster"));
  }
```

For short cases like these I think you're right, not extracting the constants does make it easier to read the test case.